src/data/metadata/MetadataD2ApiRepository.ts

```diff
@@ -247,3 +300,43 @@ const formatStats = (stats: Stats) => ({
     ..._.omit(stats, ["created"]),
     imported: stats.created,
 });
+
+// (max_size=8000 - rest_url=1000) / (id_length=11 + comma_encoded=3)
+const maxIds = 500;
+
+const defaultOrder = { field: "id", order: "asc" } as const;
+
+interface GetListAllOptions {
+    type: ListMetadataParams["type"];
+    fields: object;
+    filter: Record<string, unknown>;
+    order?: ListMetadataParams["order"];
+}
+
+interface GetListPaginatedOptions extends GetListAllOptions {
+    page?: number;
+    pageSize?: number;
+}
+
+type GetListGenericResponse =
+    | { useSingleApiRequest: false; objects: unknown[] }
+    | { useSingleApiRequest: true; order: string };
+
+function getIdFilter(
+    filter: Record<string, unknown>,
+    maxIds: number
+): { inIds: string[]; value: object } | null {
+    const inIds = filter && filter["id"] ? (filter["id"] as { in?: string[] })["in"] : undefined;
```

Comment:
```suggestion
function getIdFilter(
    filter: Dictionary<FilterValueBase>,
    maxIds: number
): { inIds: string[]; value: object } | null {
    const inIds = filter ? filter["id"] : undefined;
```

With a more significant type we can simplify the expression a lot

Reply:
Cool. I was unsure why we were using `unknown`, but I didn't want to tweak with the types, not it looks better. I've also simplified the conditional, now with better types is much more simple: b76efd19